tools/util/OperationManager.hpp

```diff
@@ -0,0 +1,1172 @@
+          ( allocator_name, rm.getAllocator(base_allocator_name),
+            advice_operation, rm.getAllocator(accessing_allocator_name),
+            std::forward<Args>(args)...);
+
+        auto allocator = new umpire::Allocator(rm.getAllocator(allocator_name));
+        this->m_alloc_array.push_back(allocator);
+      };
+    }
+  }
+
+  //
+  // FixedPool
+  //
+  template<typename... Args>
+  void bld_fixedpool(
+      const bool introspection
+    , const std::string& allocator_name
+    , const std::string& base_allocator_name
+    , Args&&... args
+  )
+  {
+    if (introspection) {
+      op = [=]() {
+        auto& rm = umpire::ResourceManager::getInstance();
+
+        rm.makeAllocator<umpire::strategy::FixedPool, true>
+          (allocator_name, rm.getAllocator(base_allocator_name), std::forward<Args>(args)...);
+
+        auto allocator = new umpire::Allocator(rm.getAllocator(allocator_name));
+        this->m_alloc_array.push_back(allocator);
+      };
+    }
+    else {
+      op = [=]() {
+        auto& rm = umpire::ResourceManager::getInstance();
+
+        rm.makeAllocator<umpire::strategy::FixedPool, false>
+          (allocator_name, rm.getAllocator(base_allocator_name), std::forward<Args>(args)...);
+
+        auto allocator = new umpire::Allocator(rm.getAllocator(allocator_name));
+        this->m_alloc_array.push_back(allocator);
+      };
+    }
+  }
+
+  //
+  // DynamicPool
+  //
+  void bld_dynamicpool(
+      const bool introspection
+    , const std::string& allocator_name
+    , const std::string& base_allocator_name
+    , const std::size_t initial_alloc_size
+    , const std::size_t min_alloc_size
+    , umpire::strategy::DynamicPool::CoalesceHeuristic /* h_fun */
+    , int alignment
+  )
+  {
+    if (introspection) {
+      op = [=]() {
+        auto& rm = umpire::ResourceManager::getInstance();
+
+        rm.makeAllocator<umpire::strategy::DynamicPool, true>
+          (   allocator_name
+            , rm.getAllocator(base_allocator_name)
+            , initial_alloc_size
+            , min_alloc_size
+            , umpire::strategy::heuristic_percent_releasable(0)
+            , alignment
+          );
+
+        auto allocator = new umpire::Allocator(
+                                rm.getAllocator(allocator_name));
+        this->m_alloc_array.push_back(allocator);
+      };
+    }
+    else {
+      op = [=]() {
+        auto& rm = umpire::ResourceManager::getInstance();
+
+        rm.makeAllocator<umpire::strategy::DynamicPool, false>
+          (   allocator_name
+            , rm.getAllocator(base_allocator_name)
+            , initial_alloc_size
+            , min_alloc_size
+            , umpire::strategy::heuristic_percent_releasable(0)
+            , alignment
+          );
+
+        auto allocator = new umpire::Allocator(
+                                rm.getAllocator(allocator_name));
+        this->m_alloc_array.push_back(allocator);
+      };
+    }
+  }
+
+  void bld_dynamicpool(
```

Comment:
I would remove this version and wherever its called in replay call it with the `umpire::strategy::heuristic_percent_releasable(0)` if not enough arguments are present.